packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -396,19 +396,29 @@ const createNetwork = () => {
       }
 
       return (e) => {
-        const functions = allFunctions(e).map((key) => e[key].bind(this));
-        listener.apply(this, [
-          {
-            ...e,
-            ...functions,
-            target: this,
-            currentTarget: this,
-            srcElement: this,
-            ports: [this],
-            source: null,
-            composedPath: () => [this],
-          },
-        ]);
+        // TODO: Should we migrate this to use a wrapper class?
```

Comment:
We could replace this entire thing with a wrapper class same as we do for responses 🤔